File: Part1/Laba_2/src/main/java/RunF2.java
public class RunF2 implements Runnable {
    private int[][] MH;
    private int[][] MK;
    private int[][] ML;
    private int[][] MG;
    Thread t;
    int n;

    public RunF2(int n) {
        t = new Thread(this, "F2");
        this.n = n;
        System.out.println("F2 started");
        t.setPriority(10);
        t.start();
    }
    @Override
    public void run() {
        MH = new int[n][n];
        MK = new int[n][n];
        ML = new int[n][n];
        Data.generateMatrix(MH);
        Data.generateMatrix(MK);
        Data.generateMatrix(ML);
        MG = Data.f2(MH, MK, ML);
        System.out.println("F2 finished");
    }

    public void printResult() {
        System.out.println("\nResult of F2: MG = MAX(MH) * (MK*ML):\n");
        Data.printMatrix(MG);
    }
}

File: Part1/Laba_3/Laba_3/RunF2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Threading;
namespace Laba_3
{
    class RunF2
    {
        public Thread thread;
        int n;
        private int[][] MH;
        private int[][] MK;
        private int[][] ML;
        private int[][] MG;

        public RunF2(int n)
        {
            thread = new Thread(this.Run);
            this.n = n;
            Console.WriteLine("F2 started");
            thread.Priority = ThreadPriority.Highest;
            thread.Start();
        }

        void Run()
        {
            MH = new int[n][];
            MK = new int[n][];
            ML = new int[n][];
            Data.GenerateMatrix(MH);
            Data.GenerateMatrix(MK);
            Data.GenerateMatrix(ML);
            MG = Data.F2(MH, MK, ML);
            Console.WriteLine("F2 finished");
        }

        public void PrintResult()
        {
            Console.WriteLine("\nResult of F2: MG = MAX(MH) * (MK*ML):\n");
            Data.PrintMatrix(MG);
        }
    }
}
